application/cms/dimension_service.py

```diff
@@ -112,16 +121,38 @@ def get_dimension_with_guid(guid):
     def delete_chart(dimension):
```

Comment:
These `delete_chart` and `delete_table` methods really feel like repeated code... but can't see any straightforward way to abstract the logic here to a single place.

Reply:
@TheDoubleK refactor this to the model?  (`Dimension().delete_chart()` and `Dimension().delete_table()`)

Also I wonder if we could make the chart and table tables do `on delete set null` so that we don't have to do this manually in code?